src/main/java/sirius/kernel/xml/XMLCall.java

```diff
@@ -128,16 +146,51 @@ public XMLStructuredOutput getOutput() throws IOException {
         return new XMLStructuredOutput(outcall.postFromOutput());
     }
 
+    private InputStream getInputStream() throws IOException {
+        if (debugLogger != null && debugLogger.isFINE()) {
+            // log the request, even when parsing fails
+            try (InputStream body = outcall.getResponse().body()) {
+                byte[] bytes = body.readAllBytes();
+                logRequest(new String(bytes, outcall.getContentEncoding()));
+                return new ByteArrayInputStream(bytes);
+            }
+        }
+        return outcall.getResponse().body();
+    }
+
+    private void logRequest(String response) throws IOException {
+        boolean isPost = outcall.isPostRequest();
```

Comment:
Better use outcall.getRequest().method There is more than one way to post. And more methods than post and get.